# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/thin-provisioning-tools/hotfix-st_rdev.patch.mips
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

src/core/device.rs:125:22
    |
125 |                 Some(metadata.st_rdev)
    |                 ---- ^^^^^^^^^^^^^^^^ expected `u64`, found `u32`
    |                 |
    |                 arguments to this enum variant are incorrect

--- ./vendor/devicemapper/src/core/device.rs.vanilla	2025-10-04 12:32:22.848423686 +0200
+++ ./vendor/devicemapper/src/core/device.rs	2025-10-04 12:35:13.706268132 +0200
@@ -122,7 +122,7 @@
     match stat::stat(path) {
         Ok(metadata) => Ok(
             if metadata.st_mode & SFlag::S_IFMT.bits() == SFlag::S_IFBLK.bits() {
-                Some(metadata.st_rdev)
+                Some(metadata.st_rdev as u64)
             } else {
                 None
             },
